src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java

```diff
@@ -92,70 +94,70 @@ void register_withEmailTaken_throwsException() {
     }
 
     @Test
-    void register_withoutUserRole_throwsError() {
+    void throwExceptionOnRegisterWithoutUserRole() {
         assertThatThrownBy(() -> userService.register(validUser))
                 .isInstanceOf(AuthenticationServiceException.class);
     }
 
     @Test
-    void register_registersUser() {
+    void registerUser() {
```

Comment:
How about `registerValidUser` (I know the name before didn't have this, but I think this is clearer)?